src/values/AbstractObjectValue.js

```diff
@@ -515,10 +516,18 @@ export default class AbstractObjectValue extends AbstractValue {
             propertyGetter = propertyShape.getGetter();
           }
         }
+        // Create an unknown array temporal
```

Comment:
Are there cases where we do not want createTemporalFromBuildFunction to return a value that is the closest match to its type argument? If not, it seems to me that ArrayValue.createTemporalWithWidenedNumericProperty should be invoked via createTemporalFromBuildFunction and never directly.

Reply:
Maybe we can at a future point, I don't feel this PR should change all those call sites though, there are quite a fair few of them.